### ISSUE TITLE
NuGet restore build step does not handle framework conditions around PackageReference

### ISSUE THREAD

**Issue by roryprimrose** [Microsoft/vsts-tasks#4264]
Hi there,

I'm having issues with a project that I've converted to multi-target using the new project format. The project is configured as ```<TargetFrameworks>netstandard1.5;net452</TargetFrameworks>```. It also pulls in a NuGet package that only supports the full framework. 

```
  <ItemGroup Condition="'$(TargetFramework)' == 'net452'">
    <PackageReference Include="SharpRaven">
      <Version>2.2.0</Version>
    </PackageReference>
  </ItemGroup>
```

This is fine in Visual Studio at compile time because the packages are restored correctly and usages of that library in code are surrounded by ```#if NET452``` blocks. The NuGet build step fails to restore this however because that package does not support netstandard1.5.

```
2017-05-08T06:55:32.0060718Z ##[section]Starting: NuGet restore **/*.sln
2017-05-08T06:55:32.0065715Z ==============================================================================
2017-05-08T06:55:32.0065715Z Task         : NuGet Restore
2017-05-08T06:55:32.0065715Z Description  : Restores NuGet packages in preparation for a Visual Studio Build step.
2017-05-08T06:55:32.0065715Z Version      : 1.0.0
2017-05-08T06:55:32.0065715Z Author       : Microsoft Corporation
2017-05-08T06:55:32.0065715Z Help         : [More Information](https://go.microsoft.com/fwlink/?LinkID=613747)
2017-05-08T06:55:32.0065715Z ==============================================================================
2017-05-08T06:55:32.4735856Z [command]C:\WINDOWS\system32\chcp.com 65001
2017-05-08T06:55:32.4815863Z Active code page: 65001
2017-05-08T06:55:32.6180986Z Found tool in cache: NuGet 4.0.0 x64
2017-05-08T06:55:32.6230981Z Resolved from tool cache: 4.0.0
2017-05-08T06:55:32.6230981Z Using version: 4.0.0
2017-05-08T06:55:32.6230981Z Found tool in cache: NuGet 4.0.0 x64
2017-05-08T06:55:32.6440983Z Detected NuGet version 4.0.0.2283 / 4.0.0
2017-05-08T06:55:32.6470985Z SYSTEMVSSCONNECTION exists true
2017-05-08T06:55:35.2411437Z save nuget.config to temp config file
2017-05-08T06:55:35.2451452Z Preparing to set credentials in NuGet.config
2017-05-08T06:55:35.2461602Z [command]C:\B\_tool\NuGet\4.0.0\x64\nuget.exe sources Remove -NonInteractive -Name myfeed -ConfigFile C:\B\4\Nuget\tempNuGet_1087.config
2017-05-08T06:55:35.5601951Z Unable to find any package source(s) matching name: myfeed.
2017-05-08T06:55:35.5611946Z [command]C:\B\_tool\NuGet\4.0.0\x64\nuget.exe sources Remove -NonInteractive -Name NuGetOrg -ConfigFile C:\B\4\Nuget\tempNuGet_1087.config
2017-05-08T06:55:35.8873748Z Unable to find any package source(s) matching name: NuGetOrg.
2017-05-08T06:55:35.8873748Z Setting credentials in NuGet.config
2017-05-08T06:55:35.8888709Z [command]C:\B\_tool\NuGet\4.0.0\x64\nuget.exe sources Add -NonInteractive -Name myfeed -Source https://myproduct.pkgs.visualstudio.com/_packaging/myfeed/nuget/v3/index.json -ConfigFile C:\B\4\Nuget\tempNuGet_1087.config
2017-05-08T06:55:36.2169079Z Package Source with Name: myfeed added successfully.
2017-05-08T06:55:36.2184074Z [command]C:\B\_tool\NuGet\4.0.0\x64\nuget.exe sources Add -NonInteractive -Name NuGetOrg -Source https://api.nuget.org/v3/index.json -ConfigFile C:\B\4\Nuget\tempNuGet_1087.config
2017-05-08T06:55:36.5564111Z Package Source with Name: NuGetOrg added successfully.
2017-05-08T06:55:36.5574112Z [command]C:\B\_tool\NuGet\4.0.0\x64\nuget.exe restore C:\B\4\s\MyProduct.Core.sln -NonInteractive -ConfigFile C:\B\4\Nuget\tempNuGet_1087.config
2017-05-08T06:55:36.9095033Z MSBuild auto-detection: using msbuild version '15.1.1012.6693' from 'C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\15.0\bin'.
2017-05-08T06:55:37.8820804Z Restoring packages for C:\B\4\s\MyProduct.Core\MyProduct.Core.csproj...
2017-05-08T06:55:37.8820804Z Restoring packages for C:\B\4\s\MyProduct.Core.UnitTests\MyProduct.Core.UnitTests.csproj...
2017-05-08T06:55:37.8820804Z Restoring packages for C:\B\4\s\MyProduct.Core.IntegrationTests\MyProduct.Core.IntegrationTests.csproj...
2017-05-08T06:55:38.2335937Z   CACHE https://api.nuget.org/v3-flatcontainer/system.componentmodel.typeconverter/index.json
2017-05-08T06:55:38.2535938Z   CACHE https://api.nuget.org/v3-flatcontainer/system.componentmodel.typeconverter/4.1.0/system.componentmodel.typeconverter.4.1.0.nupkg
2017-05-08T06:55:38.2695933Z   GET https://myproduct.pkgs.visualstudio.com/_packaging/myfeed/nuget/v3/flat2/system.componentmodel.typeconverter/index.json
2017-05-08T06:55:38.6892499Z Package SharpRaven 2.2.0 is not compatible with netstandard1.5 (.NETStandard,Version=v1.5). Package SharpRaven 2.2.0 supports:
2017-05-08T06:55:38.6897499Z   - net35 (.NETFramework,Version=v3.5)
2017-05-08T06:55:38.6897499Z   - net40 (.NETFramework,Version=v4.0)
2017-05-08T06:55:38.6897499Z   - net45 (.NETFramework,Version=v4.5)
2017-05-08T06:55:38.6897499Z One or more packages are incompatible with .NETStandard,Version=v1.5.
2017-05-08T06:55:38.7517577Z Committing restore...
2017-05-08T06:55:38.7577573Z Generating MSBuild file C:\B\4\s\MyProduct.Core\obj\MyProduct.Core.csproj.nuget.g.props.
2017-05-08T06:55:38.7657573Z Generating MSBuild file C:\B\4\s\MyProduct.Core\obj\MyProduct.Core.csproj.nuget.g.targets.
2017-05-08T06:55:38.7667670Z Writing lock file to disk. Path: C:\B\4\s\MyProduct.Core\obj\project.assets.json
2017-05-08T06:55:38.7827591Z CredentialProvider.TeamBuild: "C:\B\_tasks\NuGetRestore_333b11bd-d341-40d9-afcf-b32d5ce6f23b\1.0.0\node_modules\nuget-task-common\NuGet\CredentialProvider\CredentialProvider.TeamBuild.exe" -uri https://myproduct.pkgs.visualstudio.com/_packaging/myfeed/nuget/v3/index.json -nonInteractive
2017-05-08T06:55:38.7837579Z CredentialProvider.TeamBuild: URI Prefixes:
2017-05-08T06:55:38.7837579Z CredentialProvider.TeamBuild:     https://myproduct.visualstudio.com/
2017-05-08T06:55:38.7837579Z CredentialProvider.TeamBuild:     https://myproduct.pkgs.visualstudio.com/
2017-05-08T06:55:38.7837579Z CredentialProvider.TeamBuild: URI: https://myproduct.pkgs.visualstudio.com/_packaging/myfeed/nuget/v3/index.json
2017-05-08T06:55:38.7837579Z CredentialProvider.TeamBuild: Is retry: False
2017-05-08T06:55:38.7837579Z CredentialProvider.TeamBuild: Matched prefix: https://myproduct.pkgs.visualstudio.com/
2017-05-08T06:55:38.8227597Z Restore failed in 877.34 ms for C:\B\4\s\MyProduct.Core\MyProduct.Core.csproj.
2017-05-08T06:55:41.1141393Z   NotFound https://myproduct.pkgs.visualstudio.com/_packaging/myfeed/nuget/v3/flat2/system.componentmodel.typeconverter/index.json 2844ms
2017-05-08T06:55:41.1221422Z   GET https://myproduct.pkgs.visualstudio.com/_packaging/myfeed/nuget/v3/flat2/system.componentmodel.typeconverter/index.json
2017-05-08T06:55:41.3808924Z   NotFound https://myproduct.pkgs.visualstudio.com/_packaging/myfeed/nuget/v3/flat2/system.componentmodel.typeconverter/index.json 258ms
2017-05-08T06:55:41.9951708Z Package SharpRaven 2.2.0 is not compatible with netcoreapp1.1 (.NETCoreApp,Version=v1.1). Package SharpRaven 2.2.0 supports:
2017-05-08T06:55:41.9951708Z   - net35 (.NETFramework,Version=v3.5)
2017-05-08T06:55:41.9951708Z   - net40 (.NETFramework,Version=v4.0)
2017-05-08T06:55:41.9956705Z   - net45 (.NETFramework,Version=v4.5)
2017-05-08T06:55:41.9956705Z One or more packages are incompatible with .NETCoreApp,Version=v1.1.
2017-05-08T06:55:42.1671709Z Committing restore...
2017-05-08T06:55:42.1671709Z Generating MSBuild file C:\B\4\s\MyProduct.Core.UnitTests\obj\MyProduct.Core.UnitTests.csproj.nuget.g.props.
2017-05-08T06:55:42.1686707Z Generating MSBuild file C:\B\4\s\MyProduct.Core.UnitTests\obj\MyProduct.Core.UnitTests.csproj.nuget.g.targets.
2017-05-08T06:55:42.1701773Z Writing lock file to disk. Path: C:\B\4\s\MyProduct.Core.UnitTests\obj\project.assets.json
2017-05-08T06:55:42.1871707Z Restore failed in 4.29 sec for C:\B\4\s\MyProduct.Core.UnitTests\MyProduct.Core.UnitTests.csproj.
2017-05-08T06:55:42.7797297Z Package SharpRaven 2.2.0 is not compatible with netcoreapp1.1 (.NETCoreApp,Version=v1.1). Package SharpRaven 2.2.0 supports:
2017-05-08T06:55:42.7797297Z   - net35 (.NETFramework,Version=v3.5)
2017-05-08T06:55:42.7797297Z   - net40 (.NETFramework,Version=v4.0)
2017-05-08T06:55:42.7802300Z   - net45 (.NETFramework,Version=v4.5)
2017-05-08T06:55:42.7802300Z One or more packages are incompatible with .NETCoreApp,Version=v1.1.
2017-05-08T06:55:42.9732310Z Committing restore...
2017-05-08T06:55:42.9732310Z Generating MSBuild file C:\B\4\s\MyProduct.Core.IntegrationTests\obj\MyProduct.Core.IntegrationTests.csproj.nuget.g.props.
2017-05-08T06:55:42.9747305Z Generating MSBuild file C:\B\4\s\MyProduct.Core.IntegrationTests\obj\MyProduct.Core.IntegrationTests.csproj.nuget.g.targets.
2017-05-08T06:55:42.9757321Z Writing lock file to disk. Path: C:\B\4\s\MyProduct.Core.IntegrationTests\obj\project.assets.json
2017-05-08T06:55:42.9937302Z Restore failed in 5.1 sec for C:\B\4\s\MyProduct.Core.IntegrationTests\MyProduct.Core.IntegrationTests.csproj.
2017-05-08T06:55:42.9977309Z 
2017-05-08T06:55:42.9977309Z Errors in C:\B\4\s\MyProduct.Core\MyProduct.Core.csproj
2017-05-08T06:55:42.9982306Z     Package SharpRaven 2.2.0 is not compatible with netstandard1.5 (.NETStandard,Version=v1.5). Package SharpRaven 2.2.0 supports:
2017-05-08T06:55:42.9982306Z       - net35 (.NETFramework,Version=v3.5)
2017-05-08T06:55:42.9982306Z       - net40 (.NETFramework,Version=v4.0)
2017-05-08T06:55:42.9982306Z       - net45 (.NETFramework,Version=v4.5)
2017-05-08T06:55:42.9987307Z     One or more packages are incompatible with .NETStandard,Version=v1.5.
2017-05-08T06:55:42.9987307Z 
2017-05-08T06:55:42.9987307Z Errors in C:\B\4\s\MyProduct.Core.UnitTests\MyProduct.Core.UnitTests.csproj
2017-05-08T06:55:42.9987307Z     Package SharpRaven 2.2.0 is not compatible with netcoreapp1.1 (.NETCoreApp,Version=v1.1). Package SharpRaven 2.2.0 supports:
2017-05-08T06:55:42.9987307Z       - net35 (.NETFramework,Version=v3.5)
2017-05-08T06:55:42.9987307Z       - net40 (.NETFramework,Version=v4.0)
2017-05-08T06:55:42.9987307Z       - net45 (.NETFramework,Version=v4.5)
2017-05-08T06:55:42.9987307Z     One or more packages are incompatible with .NETCoreApp,Version=v1.1.
2017-05-08T06:55:42.9987307Z 
2017-05-08T06:55:42.9987307Z Errors in C:\B\4\s\MyProduct.Core.IntegrationTests\MyProduct.Core.IntegrationTests.csproj
2017-05-08T06:55:42.9992320Z     Package SharpRaven 2.2.0 is not compatible with netcoreapp1.1 (.NETCoreApp,Version=v1.1). Package SharpRaven 2.2.0 supports:
2017-05-08T06:55:42.9992320Z       - net35 (.NETFramework,Version=v3.5)
2017-05-08T06:55:42.9992320Z       - net40 (.NETFramework,Version=v4.0)
2017-05-08T06:55:42.9992320Z       - net45 (.NETFramework,Version=v4.5)
2017-05-08T06:55:42.9992320Z     One or more packages are incompatible with .NETCoreApp,Version=v1.1.
2017-05-08T06:55:42.9992320Z 
2017-05-08T06:55:42.9992320Z NuGet Config files used:
2017-05-08T06:55:42.9992320Z     C:\B\4\Nuget\tempNuGet_1087.config
2017-05-08T06:55:42.9992320Z 
2017-05-08T06:55:42.9992320Z Feeds used:
2017-05-08T06:55:42.9992320Z     https://myproduct.pkgs.visualstudio.com/_packaging/myfeed/nuget/v3/index.json
2017-05-08T06:55:42.9992320Z     https://api.nuget.org/v3/index.json
2017-05-08T06:55:43.0272362Z ##[error]Error: C:\B\_tool\NuGet\4.0.0\x64\nuget.exe failed with return code: 1
2017-05-08T06:55:43.0272362Z ##[error]Packages failed to restore
2017-05-08T06:55:43.0292336Z ##[section]Finishing: NuGet restore **/*.sln
```

The current workaround is to allow that step to continue on error and, for good measure, re-enable the nuget restore as part of the msbuild step.

**Comment by roryprimrose** [comment]
Forgot to add, this is using the v1 of the build task

**Comment by chrisrpatterson** [comment]
What version of the NuGet package manager are you using in VS?  If you used that same version of the NuGet command line doe sit have an issue?

**Comment by roryprimrose** [comment]
@chrisrpatterson v1 of the build task no longer provides this as an option, but you can see from the build log that it has resolved v4.0 of the NuGet CLI

```
2017-05-08T06:55:32.6180986Z Found tool in cache: NuGet 4.0.0 x64
2017-05-08T06:55:32.6230981Z Resolved from tool cache: 4.0.0
2017-05-08T06:55:32.6230981Z Using version: 4.0.0
2017-05-08T06:55:32.6230981Z Found tool in cache: NuGet 4.0.0 x64
2017-05-08T06:55:32.6440983Z Detected NuGet version 4.0.0.2283 / 4.0.0
```

**Comment by alexmullans** [comment]
You'll most likely need NuGet 4.1 for this. You can make that available in your build using the new NuGet Tool Installer, which will show up in the task catalog in a week or so as part of the next deployment.
